zerver/outgoing_webhooks/generic.py

```diff
@@ -0,0 +1,34 @@
+from __future__ import absolute_import
+from typing import Any, Dict, Text, Tuple, Callable, Mapping
+import json
+from requests import Response
+from zerver.models import UserProfile
+
+from zerver.lib.outgoing_webhook import OutgoingWebhookServiceInterface
+
+class Generic(OutgoingWebhookServiceInterface):
```

Comment:
An instance of this class would pretty much work as a service. It would store the service specific configurations (like name, base_url, token, etc) and methods like `process_event`, etc for operations. So yeah it is a `GenericOutgoingWebhookService`.